### DISCUSSION TITLE
How to rename fields in a column containing an array of structs

### DISCUSSION THREAD

**Discussion by stephen-bowser** [ibis-project/ibis#10319] [Q&A] [answered]
After using the array [zip](https://ibis-project.org/reference/expression-collections#ibis.expr.types.arrays.ArrayValue.zip) function, it would be helpful to be able to rename the fields back to something more intuitive, since this function changes the field names to f1, f2 etc. What is the best way of doing this?
I've tried using `.map` combined with `.rename`, but it seems `.rename` isn't supported on the elements in the lambda function.

**Comment by cpcloud** [comment, marked as answer]
Renaming struct fields seems like a good feature request. An implementation might look something like constructing a new struct with the desired names, using the fields of the to-be-renamed-struct.
